### DISCUSSION TITLE
Python console may not show new print until resized

### DISCUSSION THREAD

**Discussion by Evernow** [pyinstaller/pyinstaller#6673] [Help] [answered]
Hello.
I am making a Python pyinstaller executable with a console on Windows that prints new output and once in a while asks for input from user, however sometimes when the user does not have the console in focus the console does not print the new output until the user resizes it or presses some input. I have a video showcasing it here:

https://user-images.githubusercontent.com/44484725/158049535-6546ddda-31c8-4870-87fe-c968992a875f.mp4

Points of note:

Using os.system('mode con: cols=80 lines=40') to resize the window, not sure if related.

It doesn't always happen, in fact the above video it took me some 20 tries to get it to be reproduced.

Source code can be found here: https://github.com/Evernow/AutoDDU_CLI/blob/main/AutoDDU_CLI.py

It is built on Windows Server 2022 with `pyinstaller --noconfirm --onefile --console --python-option "u" --icon chikabest.ico --uac-admin AutoDDU_CLI.py` 

Any help is greatly appreciated!

**Comment by rokm** [comment, marked as answer]
Is that full square above the left eye text selection cursor?

Because if you select text in Windows console, it pauses the process until you clear the cursor, which I think matches what you are describing here. Incidentally, resizing console clears the selection cursor. (And I guess it took 20 tries to accidentally select the text in console).

**Reply by Evernow** [reply]
> Is that full square above the left eye text selection cursor?

Indeed

> Because if you select text in Windows console, it pauses the process until you clear the cursor, which I think matches what you are describing here. 

Thank you for this bit of information, is there a way to avoid this from happening?

**Reply by rokm** [reply]
You can probably disable Quick Edit by calling [`SetConsoleMode`](https://docs.microsoft.com/en-us/windows/console/setconsolemode) via `ctypes` at the start of your program; see [here](https://stackoverflow.com/questions/37500076/how-to-enable-windows-console-quickedit-mode-from-python) for examples. 

This will prevent accidental selection when user clicks in the console window; they'll have to use Edit -> Mark to activate selection. Activating selection will still pause the output (and the program execution once the output buffer gets full), though - not sure if there's any way around that.

**Reply by Evernow** [reply]
> You can probably disable Quick Edit by calling [`SetConsoleMode`](https://docs.microsoft.com/en-us/windows/console/setconsolemode) via `ctypes` at the start of your program; see [here](https://stackoverflow.com/questions/37500076/how-to-enable-windows-console-quickedit-mode-from-python) for examples.
> 
> This will prevent accidental selection when user clicks in the console window; they'll have to use Edit -> Mark to activate selection. Activating selection will still pause the output (and the program execution once the output buffer gets full), though - not sure if there's any way around that.

Apologies I didn't reply to you. But I just want to thank you. You were indeed correct and saved me a lot of headaches.
